Sources/SwiftSyntax/SourceLocation.swift

```diff
@@ -95,6 +146,15 @@ public final class SourceLocationConverter {
   /// Position at end of file.
   let endOfFile: AbsolutePosition
 
+  /// The information from all `#sourceLocation` directives in the file
+  /// necessary to compute presumed locations.
+  ///
+  /// - `sourceLine` is the line at which the `#sourceLocation` statement occurs
+  ///   within the current file.
+  /// - `fileArgument` is the `file` argument of the `#sourceLocation` directive.
+  /// - `lineArgument` is the `line` argument of the `#sourceLocation` directive.
+  var sourceLocationDirectives: [(sourceLine: Int, arguments: (file: String, line: Int)?)] = []
```

Comment:
Could you add a struct for this? Named tuples are okay but a nested named tuple is pushing it IMO 😅